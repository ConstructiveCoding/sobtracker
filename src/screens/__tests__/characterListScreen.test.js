import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import CharacterListScreen from '../characterListScreen';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Character List Screen', () => {
  it('should render all of the characters', () => {
    const initialState = {
      characters: [{}],
    };
    const store = mockStore(initialState);

    const tree = renderer.create(
      <CharacterListScreen
        store={store}
        provider={{}}
        navigation={{ setParams: () => {} }}
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
