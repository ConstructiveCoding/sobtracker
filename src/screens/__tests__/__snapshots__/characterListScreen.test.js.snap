// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character List Screen should render all of the characters 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
